assets/pages/oneoff-contributions/components/oneoffInlineContributionsPayment.jsx

```diff
@@ -0,0 +1,178 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+import { connect } from 'react-redux';
+import type { Dispatch } from 'redux';
+
+import StripeInlineForm from 'components/stripeInlineForm/stripeInlineForm';
+import PayPalContributionButton from 'containerisableComponents/payPalContributionButton/payPalContributionButton';
+import ErrorMessage from 'components/errorMessage/errorMessage';
+
+import { validateEmailAddress } from 'helpers/utilities';
+
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import type { ReferrerAcquisitionData } from 'helpers/tracking/acquisitions';
+import type { Participations } from 'helpers/abTests/abtest';
+import type { Currency } from 'helpers/internationalisation/currency';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import type { Status } from 'helpers/switch';
+import { stripeInlineFormActionsFor } from 'components/stripeInlineForm/stripeInlineFormActions';
+
+import { checkoutError, type Action } from '../oneoffContributionsActions';
+import postCheckout from '../helpers/ajax';
+
+
+// ----- Types ----- //
+
+type PropTypes = {
+  dispatch: Function,
+  email: string,
+  error: ?string,
+  isFormEmpty: boolean,
+  amount: number,
+  referrerAcquisitionData: ReferrerAcquisitionData,
+  isoCountry: IsoCountry,
+  countryGroupId: CountryGroupId,
+  checkoutError: (?string) => void,
+  abParticipations: Participations,
+  currency: Currency,
+  isTestUser: boolean,
+  isPostDeploymentTestUser: boolean,
+  stripeSwitchStatus: Status,
+  payPalSwitchStatus: Status,
+  isStripeLoaded: boolean,
+  stripeIsLoaded: () => void,
+  stripeInlineErrorMessage: ?string,
+  stripeInlineSetError: (string) => void,
+  stripeInlineResetError: () => void,
+};
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state) {
+  return {
+    isTestUser: state.page.user.isTestUser || false,
+    isPostDeploymentTestUser: state.page.user.isPostDeploymentTestUser,
+    email: state.page.user.email,
+    error: state.page.oneoffContrib.error,
+    isFormEmpty: state.page.user.email === '' || state.page.user.fullName === '',
+    amount: state.page.oneoffContrib.amount,
+    referrerAcquisitionData: state.common.referrerAcquisitionData,
+    isoCountry: state.common.country,
+    countryGroupId: state.common.countryGroup,
+    abParticipations: state.common.abParticipations,
+    currency: state.common.currency,
+    stripeSwitchStatus: state.common.switches.oneOffPaymentMethods.stripe,
+    payPalSwitchStatus: state.common.switches.oneOffPaymentMethods.payPal,
+    isStripeLoaded: state.page.stripeInlineForm.isStripeLoaded,
+    stripeInlineErrorMessage: state.page.stripeInlineForm.errorMessage,
+  };
+}
+
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
```

Comment:
Is `Action` the complete range of actions here? What about the Stripe `Action` type?

Could this be something like?

```jsx
Dispatch<StripeAction | OneOffCheckoutAction>
```